ios/NumberTileGame/F3HNumberTileGameViewController.m

```diff
@@ -246,9 +247,7 @@ - (void)resetButtonTapped {
 }
 
 - (void)exitButtonTapped {
-    NSLog(@"=====>");
-    NSLog(@"=====> The score of this game was %li", self.model.score);
-    NSLog(@"=====>");
+    [HighScoreArchiver addScore:@(self.model.score)];
```

Comment:
This only saves a score when leaving the game. If I play an awesome game and get 10,000 points, then start a new game and get 10 points, then exit, the game will only remember that I got 10 points. :-(

Reply:
Good point!

Reply:
You added a check in `resetButtonTapped`, so now you're recording scores when the player voluntarily resets... but now if I play an awesome game and get 10,000 points, then quit the app, my score isn't recorded!

It looks like there's a delegate method `gameFinishedWithVictory:didWin:score:` that would be the right place to record the score. `F3HNumberTileGameViewController` defines the delegate protocol but it doesn't look like anything implements it, so that would have to be created to do this properly.